### PR TITLE
Learners live on Microsoft.ML.Trainers

### PR DESCRIPTION
**Pull request by sfilipi**
Addresses part of #1318  by moving all the trainers and predictors in Microsoft.ML.Trainers. 

The rest of the work about xtensions and consistency will come in the next PR.

### CODE REVIEW

**Comment by Ivanidzo4ka** [comment]
> namespace Microsoft.ML.Runtime.Ensemble.OutputCombiners

Don't they deserve to be part of not runtime namespace?

---
Refers to: src/Microsoft.ML.Ensemble/OutputCombiners/MultiStacking.cs:24 in 5befeae. [](commit_id = 5befeae76223e597ef410218587bbc31e7eddaf2, deletion_comment = False)

**Comment by Ivanidzo4ka** [comment]
> namespace Microsoft.ML.Runtime.Ensemble.OutputCombiners

Or we do it only for estimator/pigsty and not "internal" implementation?

---
In reply to: [432325591](https://github.com/dotnet/machinelearning/pull/1340#issuecomment-432325591) [](ancestors = 432325591)

---
Refers to: src/Microsoft.ML.Ensemble/OutputCombiners/MultiStacking.cs:24 in 5befeae. [](commit_id = 5befeae76223e597ef410218587bbc31e7eddaf2, deletion_comment = False)

File: src/Microsoft.ML.StandardLearners/Standard/Online/OnlineLearnerCatalog.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@
 using Microsoft.ML.Runtime;
 using Microsoft.ML.Runtime.Data;
 using Microsoft.ML.Runtime.Learners;
+using Microsoft.ML.Trainers;
 using System;
 
 namespace Microsoft.ML
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.StandardLearners/Standard/Online/OnlineLearnerCatalog.cs]: >Microsoft.ML [](start = 10, length = 12)

should be Trainers right?

Comment (Ivanidzo4ka) [reply src/Microsoft.ML.StandardLearners/Standard/Online/OnlineLearnerCatalog.cs]: or staticPipe

---
In reply to: [227479981](https://github.com/dotnet/machinelearning/pull/1340#discussion_r227479981) [](ancestors = 227479981)

File: src/Microsoft.ML.FastTree/FastTreeClassification.cs [hunk 2]
```diff
@@ -39,7 +40,7 @@
     "FastTree Binary Executor",
     FastTreeBinaryPredictor.LoaderSignature)]
 
-namespace Microsoft.ML.Runtime.FastTree
+namespace Microsoft.ML.Trainers
```
Comment (TomFinley) [inline src/Microsoft.ML.FastTree/FastTreeClassification.cs]: >Trainers [](start = 23, length = 8)

I could be wrong, but my (possibly flawed) understanding was that we would have the *extension*methods for the `MLContext` objects in `Microsoft.ML.Trainers`, while keeping the actual implementing classes properly encapsulated still in the subnamespaces. I would prefer that the classes for FastTree *not* live in bifurcated namespaces.

**Comment by sfilipi** [comment]
Closing this per the discussion above, and proceeding to adding xtensions on MLContext live in that namespace.

**Pull request closed without merging**